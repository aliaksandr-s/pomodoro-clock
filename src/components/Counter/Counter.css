.counter {
  font-size: 1.5rem;
  user-select: none;
}

.counter__value {
  margin: 0 5px;
}

.counter__controll {
  cursor: pointer;
}