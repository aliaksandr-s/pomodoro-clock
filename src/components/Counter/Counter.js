// @flow

import React from 'react'
import './Counter.css'

type Props = {
  onDecrease: () => void,
  onIncrease: () => void,
  value: number
}

const Counter = (props: Props): React.Element<*> => (
  <div className="counter">
    <span className="counter__controll" role="presentation" onClick={props.onDecrease}>-</span>
    <span className="counter__value">{props.value}</span>
    <span className="counter__controll" role="presentation" onClick={props.onIncrease}>+</span>
  </div>
  )

export default Counter
