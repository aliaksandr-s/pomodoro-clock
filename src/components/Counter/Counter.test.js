import React from 'react'
import render from 'react-test-renderer'
import Counter from './Counter'

it('renders correctly', () => {
  const rendered = render.create(
    <Counter value={10} />,
  )
  expect(rendered.toJSON()).toMatchSnapshot()
})

