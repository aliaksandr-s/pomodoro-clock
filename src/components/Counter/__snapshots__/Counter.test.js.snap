// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="counter"
>
  <span
    className="counter__controll"
    onClick={undefined}
    role="presentation"
  >
    -
  </span>
  <span
    className="counter__value"
  >
    10
  </span>
  <span
    className="counter__controll"
    onClick={undefined}
    role="presentation"
  >
    +
  </span>
</div>
`;
